demo/examples/src/wallet_connect.cc

```diff
@@ -153,6 +163,52 @@ int main(int argc, char *argv[]) {
                       << bytes_to_hex_string(receipt).c_str() << std::endl;
         }
 
+        // send contract transaction
+        if (test_erc20) {
+            WalletConnectTxCommon common;
+            // Verify the contract
+            // Test ERC20 Token: GLD
+            // https://testnet.cronoscan.com/token/0xc213a7b37f4f7ec81f78895e50ea773aa8e78255
+            Erc20 erc20 =
+                new_erc20("0xC213a7B37F4f7eC81f78895E50EA773aA8E78255",
+                          "https://evm-t3.cronos.org", 338);
+            assert(erc20.name() == "Gold");
+            assert(erc20.symbol() == "GLD");
+            assert(erc20.decimals() == 18);
+            rust::String from_address = rust::String(
+                std::string("0x") +
+                address_to_hex_string(result.addresses[0].address).c_str());
+            U256 erc20_balance = erc20.balance_of(from_address);
+            std::cout << "erc20 balance=" << erc20_balance.to_string()
+                      << std::endl;
+
+            // construct tx info
+            rust::String contract_action =
+                R"({
+                      "Erc20Transfer": {
+                        "contract_address": "0xC213a7B37F4f7eC81f78895E50EA773aA8E78255",
+                        "to_address": "0xA914161b1b8d9dbC9c5310Fc7EBee5A5B18044b7",
+                        "amount": "1"
+                      }
+                   })";
+
+            common.chainid = result.chain_id;
+            common.web3api_url =
+                "https://evm-t3.cronos.org"; // TODO unnessary for
+                                             // walletconnect
+
+            rust::Vec<uint8_t> tx_hash = client->send_contract_transaction(
+                contract_action, common, result.addresses[0].address);
+
+            std::cout << "transaction_hash="
+                      << bytes_to_hex_string(tx_hash).c_str() << std::endl;
+
+            // TODO verify the balance is deducted, after transaction
+            // successful
```

Comment:
maybe adding timeout to wait for blocks,
how many blocks?

Reply:
We haven't exposed the get_block_number apis. It is marked as TODO, would be fixed in the future.

Reply:
And to make sure the transaction is successful, we have to check the receipt: https://github.com/crypto-com/defi-wallet-core-rs/pull/901